rotkehlchen/tests/api/test_history_base_entry.py

```diff
@@ -507,11 +507,24 @@ def test_get_events(rotkehlchen_api_server: 'APIServer'):
             json={'group_by_event_ids': False, 'offset': 0, 'limit': 5, 'exclude_ignored_assets': exclude_ignored_assets},  # noqa: E501
         )
         result = assert_proper_response_with_result(response)
-        assert len(result['entries']) == min(found, 5)
-        assert result['entries_found'] == found
+        assert len(result['entries']) == min(sub_events_found, 5)
+        assert result['entries_found'] == events_found
         assert result['entries_limit'] == 100
         assert result['entries_total'] == 9
 
+    # test pagination works fine with/without exclude_ignored_assets filter with/without premium
+    db_history_events = DBHistoryEvents(rotkehlchen_api_server.rest_api.rotkehlchen.data.db)
+    with db_history_events.db.user_write() as cursor:
+        for limit, exclude_ignored, total, found in (
+            (9999999999, False, 6, 6), (9999999999, True, 2, 2), (1, False, 6, 1), (1, True, 2, 1),
+        ):
+            assert db_history_events.get_history_events_count(
+                cursor=cursor,
+                group_by_event_ids=True,
+                query_filter=HistoryEventFilterQuery.make(exclude_ignored_assets=exclude_ignored),
+                entries_limit=limit,
+            ) == (total, found)
```

Comment:
maybe you could create a more realistic test with some groups having a couple of events and filtering with something that is not 1? I don't understand very well this test